apps/extension/src/ui/domains/Swap/swap-modules/simpleswap-swap-module.ts

```diff
@@ -884,37 +894,35 @@ export const simpleswapSwapModule: SwapModule = {
   decentralisationScore: DECENTRALISATION_SCORE,
 }
 
-export const retryStatus = async (
-  get: Getter,
-  id: string,
-  attempts = 0,
-): Promise<{ status: Exchange["status"] }> => {
-  try {
-    const exchange = await simpleSwapSdk.getExchange(id)
+export const swapStatus$ = (id: string): Observable<Exchange["status"] | undefined> =>
+  retryStatus$(id).pipe(
+    switchMap((status) => {
+      if (status === undefined) return of(undefined)
 
-    if (!["finished", "failed"].includes(exchange.status) && exchange.code !== 401) {
-      get(refreshSwapStatusAtom)
-    }
+      const shouldRefresh = (status: Exchange["status"] | undefined) =>
+        !(status && ["finished", "failed", "expired", "refunded"].includes(status))
 
-    return { status: exchange.status }
-  } catch (cause) {
-    const error = cause as Error & { response?: { status?: number } }
-    // because we're using a broker, the tx isnt always available immediately in their api service
-    // so we wait a bit and retry
-    if (error.name === "AxiosError" && error?.response?.status === 404 && attempts < 10) {
-      await new Promise((resolve) => setTimeout(resolve, 5000))
-      return retryStatus(get, id, attempts + 1)
-    }
-    throw cause
-  }
-}
+      // refresh every 20s if status isn't final
+      if (shouldRefresh(status)) {
+        return interval(20_000).pipe(
+          startWith(-1),
+          switchMap((i) => (i === -1 ? of(status) : retryStatus$(id))),
+          takeWhile((status) => shouldRefresh(status), true),
+        )
+      }
+      return of(status)
+    }),
+  )
 
-const _refreshSwapStatusAtom = atom(0)
-const refreshSwapStatusAtom = withAtomEffect(_refreshSwapStatusAtom, (_get, set) => {
-  const intervalId = setInterval(
-    // increment _refreshSwapStatus by 1 every 20s
-    () => set(_refreshSwapStatusAtom, (i) => (i + 1) % Number.MAX_SAFE_INTEGER),
-    20_000,
+const retryStatus$ = (id: string): Observable<Exchange["status"] | undefined> =>
+  defer(() => simpleSwapSdk.getExchange(id).then((exchange) => exchange.status)).pipe(
+    // retry up to 10 times, wait 5s between each retry
+    retry({ count: 10, delay: 5_000 }),
+
+    // log when all retries failed, and return undefined
+    catchError((error) => {
+      // eslint-disable-next-line no-console
+      console.error(`Failed to fetch exchange status for '${id}'`, error)
+      return of(undefined)
```

Comment:
This is the core of the fix - we return undefined (which is rendered as `Unknown` in the UI) instead of throwing at this point!